rocky/reports/report_types/findings_report/report.py

```diff
@@ -44,33 +39,37 @@ def generate_data(self, input_ooi: str, valid_time: datetime) -> dict[str, Any]:
             reference, depth=TREE_DEPTH, types={Finding}, valid_time=valid_time
         ).store
 
-        for ooi in tree.values():
-            if ooi.ooi_type == "Finding":
-                findings.append(ooi)
+        findings = [ooi for ooi in tree.values() if ooi.ooi_type == "Finding"]
+        all_finding_types = self.octopoes_api_connector.list_objects(types={FindingType}, valid_time=valid_time).items
 
         for finding in findings:
-            try:
-                finding_type = self.octopoes_api_connector.get(Reference.from_str(finding.finding_type), valid_time)
-                severity = finding_type.risk_severity.name.lower()
-                total_by_severity[severity] += 1
+            filter_finding_type = [x for x in all_finding_types if x.id == finding.finding_type.tokenized.id]
+
+            if not filter_finding_type:
+                continue
+
+            finding_type = filter_finding_type[0]
```

Comment:
It seems to me that you're attempting to retrieve the next available finding type, but creating an entire list for just one element seems unnecessary. So there's room for improvement here:
- Use a generator for the filtering or use `filter`
- Use `next` to find the next available element in the generator
- Catch `StopIteration`, which is an indication that the generator has been read and there are no elements available anymore
So in pseudo code this would look like:
```python3
filtered_finding_type = filter(lambda x: x.id == finding.finding_type.tokenized.id, all_finding_types)
try:
    finding_type = next(filtered_finding_type)
except:
    continue
```

Reply:
Nice, thanks! Performance went from 6,84s to 4,52s!